lisa/features/infiniband.py

```diff
@@ -26,3 +58,239 @@ def is_over_sriov(self) -> bool:
+        ).is_not_empty()
+        return device_info
+
+    def _get_ib_device_names(self) -> List[str]:
+        node = self._node
+        result = node.execute(
+            "ls /sys/class/infiniband",
+            expected_exit_code=0,
+            expected_exit_code_failure_message="Failed to get InfiniBand"
+            " devices from /sys/class/infiniband",
+        )
+
+        assert_that(result.stdout).described_as(
+            "No infiniband devices found in /sys/class/infiniband"
+        ).is_not_empty()
+        return result.stdout.split()
+
+    def setup_rdma(self) -> None:
+        node = self._node
+        # Dependencies
+        kernel = node.tools[Uname].get_linux_information().kernel_version_raw
+        ubuntu_required_packages = [
+            "build-essential",
+            "numactl",
+            "rpm",
+            "libnuma-dev",
+            "libmpc-dev",
+            "libmpfr-dev",
+            "libxml2-dev",
+            "m4",
+            "byacc",
+            "python-dev",
+            "python-setuptools",
+            "tcl",
+            "environment-modules",
+            "tk",
+            "texinfo",
+            "libudev-dev",
+            "binutils",
+            "binutils-dev",
+            "selinux-policy-dev",
+            "flex",
+            "libnl-3-dev",
+            "libnl-route-3-dev",
+            "libnl-3-200",
+            "bison",
+            "libnl-route-3-200",
+            "gfortran",
+            "cmake",
+            "libnl-3-dev",
+            "libnl-route-3-dev",
+            "libsecret-1-0",
+            "dkms",
+            "python-setuptools",
+        ]
+        redhat_required_packages = [
+            "git",
+            "zip",
+            "python3",
+            "kernel-rpm-macros",
+            "gdb-headless",
+            "python36-devel",
+            "elfutils-libelf-devel",
+            "rpm-build",
+            "make",
+            "gcc",
+            "tcl",
+            "tk",
+            "gcc-gfortran",
+            "tcsh",
+            "kernel-devel",
+            "kernel-modules-extra",
+            "createrepo",
+            "libtool",
+            "fuse-libs",
+        ]
+        if isinstance(node.os, CentOs):
+            node.execute(
+                "yum install -y https://partnerpipelineshare.blob.core.windows.net"
+                f"/kernel-devel-rpms/kernel-devel-{kernel}.rpm",
+                sudo=True,
+            )
+
+        if isinstance(node.os, Redhat):
+            if node.os.information.version.major == 7:
+                redhat_required_packages.append("python-devel")
+            else:
+                redhat_required_packages.append("python3-devel")
+                redhat_required_packages.append("python2-devel")
+            node.os.install_packages(list(redhat_required_packages))
+        elif isinstance(node.os, Ubuntu):
+            node.os.install_packages(list(ubuntu_required_packages))
+        else:
+            raise SkippedException(
```

Comment:
Let's make it failed so far. We may support them when reviewing the failure. If we decide not to support it in some distro, then use skipped.